runtime/js/40_testing.js

```diff
@@ -792,6 +928,47 @@
     globalThis.console = originalConsole;
   }
 
+  async function runBenchmarks({
+    filter = null,
+  } = {}) {
+    core.setMacrotaskCallback(handleOpSanitizerDelayMacrotask);
+
+    const origin = getTestOrigin();
+    const originalConsole = globalThis.console;
+
+    globalThis.console = new Console(reportTestConsoleOutput);
+
+    const only = ArrayPrototypeFilter(benches, (bench) => bench.only);
+    const filtered = ArrayPrototypeFilter(
+      only.length > 0 ? only : benches,
+      createTestFilter(filter),
+    );
+
+    reportTestPlan({
+      origin,
+      total: filtered.length,
+      filteredOut: benches.length - filtered.length,
+      usedOnly: only.length > 0,
+    });
+
+    for (const bench of filtered) {
+      const description = {
+        origin,
+        name: bench.name,
+      };
+      const earlier = DateNow();
+
+      reportTestWait(description);
+
+      const result = await runBench(bench, description);
+      const elapsed = DateNow() - earlier;
```

Comment:
I really wish this could be done in Rust using `Instant` and V8 APIs for best results.

Reply:
Measurements are now taken in Rust